pay-api/src/pay_api/services/eft_service.py

```diff
@@ -35,5 +43,58 @@ def create_invoice(self, payment_account: PaymentAccount, line_items: [PaymentLi
         # Do nothing here as the invoice references will be created later for eft payment reconciliations (TDI17).
 
     def apply_credit(self, invoice: Invoice) -> None:
-        """Apply credit to the invoice."""
-        self._release_payment(invoice=invoice)
+        """Apply eft credit to the invoice."""
+        invoice_balance = invoice.total - (invoice.paid or 0)  # balance before applying credits
+        payment_account = PaymentAccount.find_by_id(invoice.payment_account_id)
+        invoice_model = InvoiceModel.find_by_id(invoice.id)
+
+        payment_account.deduct_eft_credit(invoice_model)
+        new_invoice_balance = invoice.total - (invoice.paid or 0)
+
+        payment = self.create_payment(payment_account=payment_account,
+                                      invoice=invoice_model,
+                                      payment_date=datetime.now(),
+                                      paid_amount=invoice_balance - new_invoice_balance).save()
+
+        self.create_invoice_reference(invoice=invoice_model, payment=payment).save()
+        self.create_receipt(invoice=invoice_model, payment=payment).save()
+
+        if invoice.invoice_status_code == InvoiceStatus.PAID.value:
+            self._release_payment(invoice=invoice)
```

Comment:
I think we want to release the payment even if it's not PAID? as per requirements for John? Otherwise the user could potentially be waiting for this queue message to unlock an NR or LEAR filing?